### ISSUE TITLE
Closing the last pane with `<ctrl-g> x` always hangs, also sometime exiting mosaic with `<ctrl-g> q` hangs

### ISSUE THREAD

**Issue by henil** [mosaic-org/mosaic#144]
Steps to reproduce:
1. Open mosaic
2. Try closing pane with `<ctrl-g> x` (it hangs, Notice this only happens when there is only one pane left, excluding the status bar pane)
3. Also sometimes exiting `<ctrl-g> q` hangs.

**Comment by TheLostLambda** [comment]
As was recommended on the Discord earlier, I'll leave this tidbit here!

When we do go to solve this problem (which I suspect is caused by one of the threads hanging on exit), I think that we should send the `Quit` messages one at a time in the same `for` loop that the threads are currently joined in. This will give us a defined shutdown order, which should make these sorts of things a little easier to reason about.

In other words, we should try to move [the `Quit` lines here](https://github.com/mosaic-org/mosaic/blob/e6a7ed31f67a9c2170b5f4a0a37ae6089b0daa4b/src/main.rs#L659) so that they are [quit one at a time here](https://github.com/mosaic-org/mosaic/blob/e6a7ed31f67a9c2170b5f4a0a37ae6089b0daa4b/src/main.rs#L684).

**Comment by imsnif** [comment]
Fixed (including the predictable closing order)

**Comment by henil** [comment]
Closed by #150
